transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -848,10 +848,6 @@ public void run() {
         public final void beginRead() {
             assertEventLoop();
 
-            if (!isActive()) {
-                return;
-            }
```

Comment:
What's the expected behavior when a user calls `ctx.read()` on a registered but not yet active channel? This changes `ctx.read()` triggers `doBeginRead()`, which may cause a different behavior. How about adding a test case so we make sure the behavior is consistent at least from now on?

Reply:
It should be pending until the channel is active, and then behave as if you called `ctx.read` at that moment.
I'll try to add a test. Do you have a suggestion for a good place to add it?

Reply:
`AbstractChannelTest` or a new test class like `ReadOnInactiveChannelTest`?

Reply:
I also found `SocketChannelNotYetConnectedTest` which sounds relevant.

Reply:
+1 for a test

Reply:
test which runs through all transport types (epoll, kqueue, nio) 💯